test/parse_function.cpp

```diff
@@ -0,0 +1,47 @@
+#include "gtest/gtest.h"
+
+#include "../src/ast/expr_ast.hpp"
+#include "../src/ast/function_ast.hpp"
+#include "../src/ast/prototype_ast.hpp"
+#include "../src/parse.hpp"
+#include "../src/token.hpp"
+
```

Comment:
Should this also include

std::string input = "FUNCTION F_Sample : INT"
                        "  VAR_INPUT"
                        "    x : INT;"
                        "    y : INT;"
                        "  END_VAR"
                        "  F_Sample := x + y;"
                        "END_FUNCTION;";

Reply:
No, this test suite only tests the parsing code, not the tokenizing code. Testing "units" of code independently helps isolate where the problem is. If lexer and parser were tested together, and the test failed, it wouldn't be obvious which portion of code had an issue.